dask/utils.py

```diff
@@ -1811,3 +1811,24 @@ class cached_property(_cached_property):
     def __set__(self, instance, val):
         """Raise an error when attempting to set a cached property."""
         raise AttributeError("Can't set attribute")
+
+
+def typename(typ, short=False):
+    """Return name of type
+
+    Examples
+    --------
+    >>> from distributed import Scheduler
+    >>> typename(Scheduler)
+    'distributed.scheduler.Scheduler'
+    >>> typename(Scheduler, short=True)
+    'distributed.Scheduler'
+    """
+    try:
+        if short:
+            module, *_ = typ.__module__.split(".")
+        else:
+            module = typ.__module__
+        return module + "." + typ.__name__
+    except AttributeError:
+        return str(typ)
```

Comment:
There's already a `typename` function in this module. Can we use that instead? It looks like we might be able to use the other `typename` as-is. It doesn't have the `short=` keyword, but IIUC we're not using that anywhere anyways

Reply:
Oh interesting I hadn't spotted that! I just upstreamed the one from `distributed`. I've merged them now.

I added `short=` myself because I found a few places where this functionality was duplicated and used the short version. So this helps reduce some code duplication in other places.